lib/spack/spack/build_systems/cmake.py

```diff
@@ -92,7 +92,7 @@ class CMakePackage(spack.package_base.PackageBase):
         # https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html
         variant(
             "build_type",
-            default="RelWithDebInfo",
+            default="Release",
             description="CMake build type",
             values=("Debug", "Release", "RelWithDebInfo", "MinSizeRel"),
         )
```

Comment:
I think we should add a comment here listing the rationale for the default based on this PR.  Also a link to this PR and its discussion would be good.